#! /bin/sh /usr/share/dpatch/dpatch-run
## 05_octave_documentation.dpatch by Andrew Ross <andrewross@users.sourceforge.net>
##
## DP: Update octave documentation for octave 2.9.
## DP: (This patch was taken from the upstream SVN repository.)

@DPATCH@

--- plplot-5.8.0.orig/bindings/octave/INSTALL	2007-11-18 20:04:00.000000000 +0000
+++ plplot-5.8.0/bindings/octave/INSTALL	2007-11-30 17:40:26.000000000 +0000
@@ -1,5 +1,6 @@
 This instructions are somehow out of date. Read also the USAGE file.
 
+
 Install:
 
 plplot_octave_org.h is the main important file. It is a massaged version
@@ -37,16 +38,15 @@
 Before installing, you can test plplot_octave:
 [you must install PLplot before, this will change soon]
 
-'cd' to the source distribution directory bindings/octave
+'cd' to the directory bindings/octave within the build tree.
 
 If you have another version of plplot installed, check, after invoking
 octave and making same plots, the plplot version, typing "plgver'" (note
 the apostrophe in plgver, meaning transpose). You should get the same
 version that you have just compiled, not the installed version.
 
-Invoke octave, (the variable  LOADPATH is automatically set to
-	LOADPATH="../bindings/octave//:"
-because of the file .octaverc which exists in the tmp directory)
+Invoke octave, (the path is automatically set by the .octaverc file
+which exists in that directory).
 
 There are several Octave demos, 'p1' up to 'p20'. Try them:
 
@@ -65,7 +65,7 @@
 	warning: empty matrix found in matrix list
 
 You can avoid it setting "empty_list_elements_ok = 1;" in your
-~/.octaverc Octave startup file (see bellow). This is beeing corrected
+~/.octaverc Octave startup file (see below). This is being corrected
 in plplot_octave, meanwhile keep remembering me :-).
 
 
@@ -84,6 +84,11 @@
 2-The demos path is /usr/local/lib/plplotx.y.z/examples/octave//
 3-The path to switch between PLplot and gnuplot is /usr/local/share/octave//
 
+Note: If you are using octave2.9 then LOADPATH is obsolete and you must 
+use addpath instead. Note also that a trailing // no longer includes 
+all subdirectories. You must explicitly add them. In particular you need
+the support subdirectory of plplot_octave.
+
 If you want to test the installation with the demos, call octave and
 set the LOADPATH as in
 
@@ -166,3 +171,5 @@
 Enjoy,
 
 Joao
+
+Update for octave 2.9 by Andrew Ross.

--- plplot-5.8.0.orig/bindings/octave/USAGE	2007-11-18 20:04:00.000000000 +0000
+++ plplot-5.8.0/bindings/octave/USAGE	2007-11-30 17:40:26.000000000 +0000
@@ -1,17 +1,22 @@
 You must add the plplot_octave script installation directory to Octave
-LOADPATH *before* the standard Octave LOADPATH. The best way to do this
-is to create a .octaverc file in your home directory. By default
+path *before* the standard Octave path if you want to use plplot 
+graphics by default. The best way to do this is to create a .octaverc 
+file in your home directory. 
 
-LOADPATH ="<install-directory>/share/plplot_octave//:"
+With octave 2.1 or earlier you can use LOADPATH, e.g.
 
-If you also want to use Octave's standard plotting package, gnuplot, you
-must set LOADPATH as:
+LOADPATH = [ ":<install-directory>/share/plplot_octave//", LOADPATH ]
+
+With octave 2.9 and later, LOADPATH has been depreciated and you need to
+use addpath instead. You also need to explicitly add in any 
+subdirectories you wish to search, e.g.
 
-LOADPATH ="<install-directory>/share/plplot_octave//:\
-	<install-directory>/share/octave//:"
+addpath("<install-directory>/share/plplot_octave", ...
+        "<install-directory>/share/plplot_octave/support")
 
-and use the command "toggle_plplot_use" to switch between the two plot packages.
-Read the INSTALL file for more details.
+If you also want to use Octave's standard plotting package, gnuplot, you
+can use the command "toggle_plplot_use" to switch between the two plot 
+packages. Read the INSTALL file for more details.
 
 Then, invoke Octave and create a vector, say
 
@@ -75,7 +80,15 @@
 NOTES: - You don't need to specify vector or matrix dimensions.
        - When setting Octave variable "automatic_replot = 1", changes take
 	 effect immediately, otherwise they only take effect at the next plot.
-	 (This does not yet apply to all plot types).
+	 (This does not yet apply to all plot types and does not work with 
+         octave 2.9).
+       - With octave 2.9 you will see various spurious warnings about variables
+         which are no longer builtins. These warnings can safely be ignored.
+         Plplot is just testing to see if these variables exist as part of
+         the octave 2.1 compatibility code. To suppress these warnings add
+         warning("off","Octave:built-in-variable-assignment"); 
+         to your .octaverc.
+         
 
 
 --- Plot window control
